### PR TITLE
WIP: Set pending condition on QueuedWorkload with message

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Initial implementation for #102
sets:

- The Queue doesn't exist
- The ClusterQueue doesn't exist

Still missing:

- The QW's namespace is not allowed by the ClusterQueue
- The workload was attempted for scheduling but it didn't fit.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
> @ArangoGutierrez: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-kueue-test-integration-main	[c6a4a94](https://github.com/kubernetes-sigs/kueue/commit/c6a4a940115eab8dc37eeee472de922298d2d055)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/105/pull-kueue-test-integration-main/1501571391223238656)	true	`/test pull-kueue-test-integration-main`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes-sigs&repo=kueue&pr=105). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3AArangoGutierrez). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes-sigs/kueue/issues?q=is:issue+is:open) when you hit one in your PR.

I know I know..... looking what I might be missing

**Comment by ArangoGutierrez** [comment]
Although this is still a WIP, I would like to have some eyes here @alculquicondor @ahg-g @denkensk

File: api/v1alpha1/queuedworkload_types.go [hunk 1]
```diff
@@ -88,12 +88,15 @@ type PodSet struct {
 
 // QueuedWorkloadStatus defines the observed state of QueuedWorkload
 type QueuedWorkloadStatus struct {
-	// conditions hold the latest available observations of the QueuedWorkload
+	// Current hold the latest available observations of the QueuedWorkload
 	// current state.
 	// +optional
+	Current QueuedWorkloadCondition `json:"current,omitempty"`
```
Comment (denkensk) [inline api/v1alpha1/queuedworkload_types.go:94]: Do we need to add `current` separately？ User can get it from conditions, right?

File: pkg/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -830,12 +830,13 @@ func TestEntryAssignFlavors(t *testing.T) {
 								},
 							},
 						},
-						Tolerations: []corev1.Toleration{corev1.Toleration{
-							Key:      "instance",
-							Operator: corev1.TolerationOpEqual,
-							Value:    "spot",
-							Effect:   corev1.TaintEffectNoSchedule,
-						}},
```
Comment (denkensk) [inline pkg/scheduler/scheduler_test.go:838]: revert?

**Review by denkensk — COMMENTED** [review]
Maybe we also need the message of scheduling failed? It's also the reason why queuedWorkload is pending now.

File: api/v1alpha1/queuedworkload_types.go (hunk 1 above)
Comment (ahg-g) [reply api/v1alpha1/queuedworkload_types.go:94]: +1, splitting like this is against the pattern, conditions are already timestamped.

**Comment by ArangoGutierrez** [comment]
I will work on a diff approach to this 
/close

**Pull request closed without merging**